### ISSUE TITLE
0.11 Migration: Missing requirement for `Default` on `#[reflect(ignore)]` field.

### ISSUE THREAD

**Issue by nicopap** [bevyengine/bevy-website#691]
The migration doesn't mention a fairly complex change.
So I had this struct:

```rust
#[derive(Component, Reflect)]
pub struct Hook {
    pub file_path: String,
    pub state: State,
    #[reflect(ignore)]
    pub hook: Box<dyn Fn(&EntityRef, &mut EntityCommands, &World, Entity) + Send + Sync + 'static>,
}
```

Now bevy requires a `Default` impl for `hook`, but it's not possible to provide such a thing. I guess I could replace it by a `Option<Box<_>>` but then I've something nullable where it should never be. Furthermore, there is no possible sensible defaults for this.

**Comment by MrGVSV** [comment]
Good catch. It should probably mention that the solution here is to either implement `Default` for `Box` or opt out of `FromReflect` using `#[reflect(from_reflect = false)]`.
